blackd.py

```diff
@@ -51,22 +52,42 @@ def make_app() -> web.Application:
             )
         except ValueError:
             return web.Response(status=400, text="Invalid line length header value")
-        py36 = False
         pyi = False
+        versions: Set[black.TargetVersion] = set()
         if PYTHON_VARIANT_HEADER in request.headers:
             value = request.headers[PYTHON_VARIANT_HEADER]
             if value == "pyi":
                 pyi = True
             else:
-                try:
-                    major, *rest = value.split(".")
-                    if int(major) == 3 and len(rest) > 0:
-                        if int(rest[0]) >= 6:
-                            py36 = True
-                except ValueError:
-                    return web.Response(
-                        status=400, text=f"Invalid value for {PYTHON_VARIANT_HEADER}"
-                    )
+                for version in value.split(","):
+                    tag = "cpy"
+                    if version.startswith("cpy"):
```

Comment:
I don't really like the explicit parsing here. Is it possible to try to match on all possible enum values here?

Reply:
Yes, it's not great. I tried to work with the following constraints:
- Continue accepting everything the previous code accepted (for backward compatibility). That's why there's the default if the user just put "3" or "2" as the version.
- Don't be too liberal in what I accept, because then future backward compatibility will bind us to continue accepting it.
- Avoid exposing our internal enum values too much. Users shouldn't have to care that we have a separate PyPy 3.5 version, but not one for other PyPy versions.
I'll see what I can change but I'm not sure there's much more I can do keeping these constraints.